frontend/src/libs/apiUtils.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * A collection of untilty functions for interfacing with the API
+ */
+
+const API_URL = "/api/v1";
+const FETCH_INIT = {
+    credentials: "same-origin",
+    headers: {
+        "Content-Type": "application/json"
+    }
+};
+
+/**
+ * Error thrown whenever an api call returns with `status==="error"`
+ *
+ * @class ApiError
+ * @extends {Error}
+ */
+class ApiError extends Error {
+    constructor(resp) {
+        const errorMessage = resp.message;
+        super(errorMessage);
+        Error.captureStackTrace(this, ApiFetchError);
+        this.response = resp;
+        this.status = resp.status;
+    }
+}
+
+/**
+ * Error thrown whenever an api fetch request
+ * returns with a status other than 200
+ *
+ * @class ApiFetchError
+ * @extends {ApiError}
+ */
+class ApiFetchError extends ApiError {
+    constructor(resp, path) {
+        const errorMessage = `Got status ${resp.status} ${
+            resp.statusText
+        } when fetching ${API_URL + path}`;
+        super(errorMessage);
+        Error.captureStackTrace(this, ApiFetchError);
+        this.response = resp;
+        this.status = resp.status;
+    }
+}
+
+// Ensure that `path` starts with a `/`
+function _ensurePath(path) {
+    return path.starsWith("/") ? path : "/" + path;
```

Comment:
should be 'path.startsWith' ?

Reply:
Sure should! Soon there'll actually be an api to query against ;-)